core/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -1317,6 +1317,24 @@ public CommitId flush(boolean force, boolean waitIfOngoing) throws EngineExcepti
         return new CommitId(newCommitId);
     }
 
+    @Override
+    public void trimTranslog() throws EngineException {
+        try (ReleasableLock lock = readLock.acquire()) {
```

Comment:
Would you mind renaming this to `ignored`? It keeps the IDE from complaining, and it makes it clear that the resource is intentionally unused through the scope of the try-with-resources block.